internal/provider/meta.go

```diff
@@ -281,6 +303,47 @@ func GetClient(i interface{}, meta interface{}) (*api.Client, error) {
 	return p.GetClient(), nil
 }
 
+// IsAPISupported receives a minimum version
+// that the Vault server version should be above.
+//
+// It uses the go-version package
+// to perform a semantic version comparison, and
+// returns:
+//    - a boolean describing whether the Vault
+//      server version was above the minimum version
+//
+// This function can be used to perform semantic version comparisons
+// to conditionally enable features, or to resolve any diffs in the TF
+// state based on the Vault version.
+func IsAPISupported(meta interface{}, minVersion *version.Version) bool {
```

Comment:
Since this function only supports `*ProviderMeta` we may want to panic here on the type assert

Reply:
Good call. Added a panic statement